modules/create_timepoint/php/ajaxtimepoint.class.inc

```diff
@@ -0,0 +1,309 @@
+<?php
+/**
+ * POST request methods.
+ *
+ * Used to retrieve info & create timepoint by the create_timepoint form.
+ * Ensures the user can create timepoint before processing
+ * the POST body (array) data.
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Create_Timepoint
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\create_timepoint;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  Create_Timepoint
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class AjaxTimepoint extends \NDB_Page
+{
+    public $skipTemplate = true;
+
+    /**
+     * This function will return a json file with DCCID and PSCID
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     * @throws \DatabaseException
+     * @throws \DeprecatedException
```

Comment:
Does it?

Reply:
@johnsaigle My IDE believes it's possible because the functions inside the handle(){...} can throw those exceptions. I personally don't believe it will ever happen. The reason I add those @throws tags is because my IDE shows an ugly squiggly line around the function comment if I don't add them.

Reply:
This is maybe a philosophical point about how Exceptions should be handled and annotated... we should discuss it as a team.

Reply:
@johnsaigle I'll bring it up at the next meeting. I hope these @throw tags can be added because it cleans up my IDE warnings.

Reply:
On the other hand you could configure the settings in your IDE :D

Either way we should figure this out as a team. It might end up going this way where we document every exception a function could throw including the functions it calls. However that's not how things are now.

Reply:
I also think you should configure your IDE.

Reply:
@johnsaigle @driusan I believe my IDE is correct to display these @throws warnings. I can look into disabling it for LORIS specifically but it's beneficial from my viewpoint to know the method will possibly throw an exception and even if the function inside is responsible; no matter how far down the chain of functions it goes.

Reply:
@johnsaigle @driusan I also know nobody seems to care because everybody is coding with vim.